src/Microsoft.OData.Core/ODataBatchReader.cs

```diff
@@ -333,9 +327,9 @@ protected virtual string GetCurrentGroupIdImplementation()
             string contentId,
             string groupId,
             IEnumerable<string> dependsOnRequestIds,
-            ODataFormat batchFormat)
+            bool dependsOnIdsValidationRequired)

```

Comment:
>bool dependsOnIdsValidationRequired [](start = 12, length = 35)

Can we do the same thing that we did in the writer, and pass null versus an empty list of requestIds instead of a boolean flag here, and validate the Url against the payloadUriConverter in the null dependsOnIds case?

Reply:
I thought about that but we are out of luck / by design here. The dependsOnRequestIds is always not null, and it is always derived (implicit) for Multipart/Mixed batch, and always explicit for Json batch.

---
In reply to: [160846029](https://github.com/OData/odata.net/pull/1020#discussion_r160846029) [](ancestors = 160846029)

Reply:
What would happen if we removed the dependsOnIdsValidationRequired flag and always validated the dependsOnRequestIds?  Shouldn't the implicit dependsOnRequestIds accumulated in the multipart/mixed case always be valid?

---
In reply to: [160847347](https://github.com/OData/odata.net/pull/1020#discussion_r160847347) [](ancestors = 160847347,160846029)

Reply:
In V4, since the payloadUriConverter is reset after changset, the validation (L336) will throw because implicit dependsOnIds could contain Content-IDs from preceding top-level requests --> error.

---
In reply to: [160848457](https://github.com/OData/odata.net/pull/1020#discussion_r160848457) [](ancestors = 160848457,160847347,160846029)

Reply:
Right; in 4.0, requests can only reference other requests within the same changeset, so payloadUriConverter only has the set of ids in the changeset, which is a subset of the requests that this request depends on coming before this request. Makes sense. 

---
In reply to: [160849419](https://github.com/OData/odata.net/pull/1020#discussion_r160849419) [](ancestors = 160849419,160848457,160847347,160846029)